### PR TITLE
Refactor task plugins

### LINKED ISSUE #319: refactor: reorganize and cleanup the Task Plugin restapi endpoint
Refactor the Task Plugin endpoint, following the pattern from #281 

### Definition of Done
- [x] Task plugin endpoint is refactored

### PR DESCRIPTION
**Pull request by alexb1200**
Refactor the task plugins endpoint to reflect changes to queue and remove wtforms dependency. Resolves #319 

# Changes
- remove forms models
- move errors from controller to service layer
- move single use function into create method in service layer
- remove inline type annotations
- add docstrings to service 
- cleanup docstrings in schema 
- add error if not found flag to all get methods

### CODE REVIEW

File: src/dioptra/restapi/task_plugin/service.py [hunk 1]
```diff
@@ -126,41 +168,91 @@ def delete(
...
 
     def get_all(
-        self, s3_collections_list: List[str], bucket: str = "plugins", **kwargs
-    ) -> List[TaskPlugin]:
+        self,
+        s3_collections_list: List[str],
+        bucket: str = "plugins",
+        raise_error_if_not_found: bool = False,
+        **kwargs,
+    ) -> list[TaskPlugin]:
```
Comment (lbarbMITRE) [inline src/dioptra/restapi/task_plugin/service.py:183]: List vs list

File: src/dioptra/restapi/task_plugin/service.py [hunk 2]
```diff
@@ -126,41 +168,91 @@ def delete(
...
 
     def get_all_in_collection(
-        self, collection: str, bucket: str = "plugins", **kwargs
-    ) -> List[TaskPlugin]:
+        self,
+        collection: str,
+        bucket: str = "plugins",
+        raise_error_if_not_found: bool = False,
+        **kwargs,
+    ) -> list[TaskPlugin]:
```
Comment (lbarbMITRE) [inline src/dioptra/restapi/task_plugin/service.py:223]: List vs list

File: tests/unit/restapi/task_plugin/test_model.py
[the remark this answers is not in the document]
Comment (alexb1200) [reply tests/unit/restapi/task_plugin/test_model.py:1]: This should probably be a separate PR given issue #316, and the goal of this PR is to remove the wtforms dependency and split functionality in the service layer.

File: src/dioptra/restapi/task_plugin/controller.py [hunk 3]
```diff
@@ -55,7 +53,7 @@ def __init__(self, *args, task_plugin_service: TaskPluginService, **kwargs) -> N
 
     @login_required
     @responds(schema=TaskPluginSchema(many=True), api=api)
-    def get(self) -> List[TaskPlugin]:
+    def get(self) -> list[TaskPlugin]:
```
[the remark this answers is not in the document]
Comment (alexb1200) [reply src/dioptra/restapi/task_plugin/controller.py:56]: In this [comment](https://github.com/usnistgov/dioptra/pull/341#issuecomment-1858470380) @jkglasbrenner suggested we move from `List` to `list`, `Dict` to `dict`, etc. in each endpoint. The new refactors should reflect this change.

File: src/dioptra/restapi/task_plugin/controller.py [hunk 4]
```diff
@@ -148,24 +145,15 @@ def get(self, taskPluginName: str) -> TaskPlugin:
...
-        if task_plugin is None:
-            log.error(
-                "TaskPlugin not found",
-                task_plugin_name=taskPluginName,
-                collection="dioptra_builtins",
-            )
-            raise TaskPluginDoesNotExistError
-
-        return task_plugin
+        return cast(TaskPlugin, task_plugin)
```
[the remark this answers is not in the document]
Comment (alexb1200) [reply src/dioptra/restapi/task_plugin/controller.py:156]: I believe this was due to mypy complaining, but if removing the cast still passes all the tests I'm happy to fix it!

**Comment by andrewhand** [comment]
365 Reviewed: File changes look clean. All tests passing on local deployment.

**Comment by jkglasbrenner** [comment]
I've begun my review of this. I also rebased on the latest version of `dev` and resolved the file conflicts (mostly due to moving all dependency injection config into `bootstrap.py`). So far, I was also able to consolidate the TaskPluginSchema into a single Schema class.

Will continue working through this later.

**Review by jkglasbrenner — APPROVED** [review]
I've wrapped up with this. The only other change I did was split `TaskPluginService` into two classes, `TaskPluginService` and `TaskPluginCollectionService` so that we only have `get()` and `get_all()` methods. I updated the one unit test affected by this change. Otherwise, it's passing all tests and looks good, so I'm going to merge to `dev`.

**Pull request merged**